pkg/controllers/workload_reconciler.go

```diff
@@ -212,6 +214,33 @@ func (r *WorkloadReconciler) isSupplyChainReady(supplyChain *v1alpha1.ClusterSup
 	return supplyChainReadyCondition.Status == "True"
 }
 
+func buildWorkloadResourceLabeler(owner, blueprint client.Object) realizer.ResourceLabeler {
+	return func(resource realizer.OwnerResource) templates.Labels {
+		switch blueprint.(type) {
+		case *v1alpha1.ClusterSupplyChain:
+			return templates.Labels{
+				"carto.run/workload-name":         owner.GetName(),
+				"carto.run/workload-namespace":    owner.GetNamespace(),
+				"carto.run/supply-chain-name":     blueprint.GetName(),
+				"carto.run/resource-name":         resource.Name,
+				"carto.run/template-kind":         resource.TemplateRef.Kind,
+				"carto.run/cluster-template-name": resource.TemplateRef.Name,
+			}
+		case *v1alpha1.ClusterDelivery:
```

Comment:
I noticed that in deliverable_reconciler we're doing

```go
func buildDeliverableResourceLabeler(owner, blueprint client.Object) realizer.ResourceLabeler {
        return func(resource realizer.OwnerResource) templates.Labels {
                return templates.Labels{
                        "carto.run/deliverable-name":      owner.GetName(),
                        "carto.run/deliverable-namespace": owner.GetNamespace(),
                        "carto.run/delivery-name":         blueprint.GetName(),
                        "carto.run/resource-name":         resource.Name,
                        "carto.run/template-kind":         resource.TemplateRef.Kind,
                        "carto.run/cluster-template-name": resource.TemplateRef.Name,
                }
        }
}
```

while here we're actually doing the switch on the type - is there a particular reason for such difference?

Reply:
for

```go
func MakeSupplychainOwnerResources(supplyChain *v1alpha1.ClusterSupplyChain) []OwnerResource {
	var resources []
```

we pass straight `v1alpha1.ClusterSupplyChain` in there - given that `build(Deliverable|Workload)` already includes the concrete type in the name of the function itself, wdyt of passing `*v1alpha1.(deliverable|workload)` in the args directly?

Reply:
@idoru I dont think we need the switch on type at all? the reconciler is building for Workloads/SCs only?